scene/gui/text_edit.cpp

```diff
@@ -959,15 +959,17 @@ void TextEdit::_notification(int p_what) {
 									icon->draw_rect(ci, gutter_rect, false, get_line_gutter_item_color(line, g));
 								} break;
 								case GUTTER_TYPE_CUSTOM: {
-									if (gutter.custom_draw_obj.is_valid()) {
-										Object *cdo = ObjectDB::get_instance(gutter.custom_draw_obj);
-										if (cdo) {
-											Rect2i gutter_rect = Rect2i(Point2i(gutter_offset, ofs_y), Size2i(gutter.width, row_height));
-											if (rtl) {
-												gutter_rect.position.x = size.width - gutter_rect.position.x - gutter_rect.size.x;
-											}
-											cdo->call(gutter.custom_draw_callback, line, g, Rect2(gutter_rect));
+									if (gutter.custom_draw_callback.is_valid()) {
+										Rect2i gutter_rect = Rect2i(Point2i(gutter_offset, ofs_y), Size2i(gutter.width, row_height));
+										if (rtl) {
+											gutter_rect.position.x = size.width - gutter_rect.position.x - gutter_rect.size.x;
 										}
+
+										Variant args[3] = { line, g, Rect2(gutter_rect) };
+										const Variant *argp[] = { &args[0], &args[1], &args[2] };
```

Comment:
A better way to pass args, without creating 2 arrays, is something like this
```C++
const Variant **argptrs = (const Variant **)alloca(sizeof(Variant *) * 3);
argptrs[0] = &line;
argptrs[1] = &g;
argptrs[2] = &rect;
```

Reply:
Can't seems get this to work as it needs the args to be cast to `Variant` first, which is easier said then done.

Reply:
You just need to cast
```C++
const Variant **argptrs = (const Variant **)alloca(sizeof(Variant *) * 3);
argptrs[0] = &(Variant)line;
argptrs[1] = &(Variant)g;
argptrs[2] = &(Variant)gutter_rect;
```

Reply:
Compiler complains about taking the address of a temporary object when casting like that.

Reply:
After checking how I did it somewhere else in the code, this is ok actually. You need to create a Variant variable, but you have multiple variables, so using array like this is fine.